SportsHub/SportsHub.Application/Services/CommentService.cs

```diff
@@ -19,5 +21,60 @@ public async Task<IEnumerable<Comment>> GetByArticleAsync(int id)
             return await _unitOfWork.CommentRepository.GetByArticleAsync(id)?? 
                    throw new NotFoundException( string.Format(ExceptionMessages.NotFound, ExceptionMessages.Comment));
         }
+
+        public async Task<bool> AddCommentAsync(CreateCommentDTO commentInput)
+        {
+            var userExists = _unitOfWork.UserRepository.GetById(commentInput.AuthorId) != null;
+            var articleExists = _unitOfWork.ArticleRepository.GetById(commentInput.ArticleId) != null;
+
+            if (!userExists || !articleExists)
+            {
+                return false;
+            }
+
+            var comment = new Comment()
+            {
+                Content = commentInput.Content,
+                AuthorId = commentInput.AuthorId,
+                ArticleId = commentInput.ArticleId,
+            };
+
+            await _unitOfWork.CommentRepository.AddCommentAsync(comment);
+            await _unitOfWork.SaveChangesAsync();
+
+            return true;
+        }
+
+        public async Task<bool> LikeCommentAsync(int commentId)
+        {
+            var comment = _unitOfWork.CommentRepository.GetById(commentId);
+
+            if (comment == null)
+            {
+                return false;
+            }
+
+            comment.Likes++;
+
+            await _unitOfWork.SaveChangesAsync();
+
+            return true;
+        }
+
+        public async Task<bool> DislikeCommentAsync(int commentId)
+        {
```

Comment:
can you like and dislike comments at the same time? maybe the should be some additional validation, if you liked it already then you can't like it again, and the same for dislike. and if you dislike you should check id the user has liked it, if yes then delete the like.

Reply:
something like another table, that contains UserId, CommentId and LikeStatus